### PR TITLE
feat(FX-3292): add filters to searchable artwork grid

### PR DESCRIPTION
**Pull request by gkartalis**
The type of this PR is: **Feature**

This PR resolves [FX-3292]

### Description

⚠️ Note: This work is behind a feature flag

- add sort & filter on searchable artwork grid
- add skeleton component on artwork grid

### Videos

| Android | iOS |
|---|---|
|https://user-images.githubusercontent.com/21178754/133605704-85bbecb0-7de7-4d33-aa55-d097eda9a333.mp4|https://user-images.githubusercontent.com/21178754/133605861-218952c7-28e2-4a09-94f9-e716f8152fa0.mp4|

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add filters to searchable artwork grid - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3292]: https://artsyproduct.atlassian.net/browse/FX-3292

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (add filters to searchable artwork grid - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c1edac8424f68857a55a38b1273bd77b3b5ac52e
</p>

File: src/lib/Scenes/Search2/SearchArtworksGrid.tsx [hunk 1]
```diff
@@ -1,55 +1,54 @@
...
+  const [isFilterArtworksModalVisible, setFilterArtworkModalVisible] = useState(false)
+  const handleCloseFilterArtworksModal = () => setFilterArtworkModalVisible(false)
+  const handleOpenFilterArtworksModal = () => setFilterArtworkModalVisible(true)
+  const artworksCount = viewer.artworks?.edges?.length
+  const appliedFilters = ArtworksFiltersStore.useStoreState((state) => state.appliedFilters)
+
+  const appliedFiltersCount = useMemo(
+    () => Object.values(getSelectedFiltersCounts(appliedFilters)).reduce((prev, value) => prev + value, 0),
+    [appliedFilters]
+  )
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/SearchArtworksGrid.tsx:33]: **suggestion**: Move it in a separate hook to reuse the logic

```javascript
export const useSelectedFiltersCounts = () => {
  const appliedFilters = ArtworksFiltersStore.useStoreState((state) => state.appliedFilters)
  return useMemo(
    () => Object.values(getSelectedFiltersCounts(appliedFilters)).reduce((prev, value) => prev + value, 0),
    [appliedFilters]
  )
}

...

const selectedFiltersCount = useSelectedFiltersCounts()
```

File: src/lib/Scenes/Search2/SearchArtworksGrid.tsx [hunk 2]
```diff
@@ -58,7 +57,50 @@ const SearchArtworksGrid: React.FC<SearchArtworksGridProps> = ({ viewer, relay }
...
+      <Separator />
+      {artworksCount === 0 ? (
+        <Box mb="80px" pt={1}>
+          {/* TODO: Change that with no results message for artwork grid??? Design? */}
+          <Text>{`We couldn't find anything for`}</Text>
+        </Box>
+      ) : (
+        <FlatList<ArtworkSection>
+          data={content}
+          contentContainerStyle={{ paddingTop: 20 }}
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/SearchArtworksGrid.tsx:98]: **nitpick**: 

```javascript
const space = useSpace()

...

<FlatList<ArtworkSection>
  data={content}
  contentContainerStyle={{ paddingTop: space(2) }}
  renderItem={({ item }) => item.content}
  keyExtractor={({ key }) => key}
/>
```

File: src/lib/Scenes/Search2/SearchArtworksGrid.tsx [hunk 3]
```diff
@@ -58,7 +57,50 @@ const SearchArtworksGrid: React.FC<SearchArtworksGridProps> = ({ viewer, relay }
...
+              </Text>
+              {appliedFiltersCount > 0 && (
+                <Text variant="small" color="blue100">
+                  {` ${bullet} ${appliedFiltersCount}`}
+                </Text>
+              )}
+            </Flex>
+          )}
+        />
+      </Flex>
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/SearchArtworksGrid.tsx:88]: **suggestion**: Move it to a separate reused component

File: src/lib/Scenes/Search2/containers/SearchArtworksContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                <Flex flex={1} flexDirection="row">
+                  <PlaceholderButton width={20} height={20} />
+                  <PlaceholderButton marginLeft={5} width={70} height={20} />
+                </Flex>
+              </Flex>
+              <Separator mb={2} />
+              <PlaceholderGrid />
+            </>
+          ),
+        })}
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/containers/SearchArtworksContainer.tsx:40]: 👏

**Comment by dimatretyak** [comment]
Please add an offset for the text when there is nothing to display

Simulator Screen Shot - iPhone 8 - 2021-09-16 at 16 15 01

**Review by dimatretyak — APPROVED** [review]
Nice work 🥇

**Comment by gkartalis** [comment]
> Please add an offset for the text when there is nothing to display
> 
> Simulator Screen Shot - iPhone 8 - 2021-09-16 at 16 15 01

Added this one that I copied over from `ArtistArtworks.tsx` will revisit this though with Design

Simulator Screen Shot - iPhone 8 - 2021-09-16 at 16 28 18

**Review by mdole — APPROVED** [review]
nice! don't think I have any comments to add!

**Comment by gkartalis** [comment]
#squashOnGreen

**Pull request merged**